src/utilities/getVariables.ts

```diff
@@ -74,29 +110,31 @@ export const getVariables = (figma: PluginAPI, settings: Settings) => {
       // get collection name and modes
       const { variableCollectionId } = variable
       const { name: collection, modes } = collections[variableCollectionId]
+
+      if (modes.length > 1) {
```

Comment:
@0m4r, when I comment out this if clause and the content inside it does not break. So, the leak is probably in `detectVariableReferencesInCollection`.

Reply:
(you cannot share the Figma file you run your test, right?)